lib/firmware/fsm_msg_ethereum.h

```diff
@@ -285,3 +285,34 @@ void fsm_msgEthereumSignTypedHash(const EthereumSignTypedHash *msg) {
   ethereum_typed_hash_sign(msg, node, resp);
   layoutHome();
 }
+
+void fsm_msgEthereum712TypesValues(Ethereum712TypesValues *msg) {
+  RESP_INIT(EthereumTypedDataSignature);
+
+  CHECK_INITIALIZED
+
+  CHECK_PIN
+
+  if (strlen(msg->eip712types) == 0) {
+    fsm_sendFailure(FailureType_Failure_Other, _("Invalid EIP-712 types property string"));
+    return;
+  }
+
+  const HDNode *node = fsm_getDerivedNode(SECP256K1_NAME, msg->address_n,
+                                          msg->address_n_count, NULL);
+  if (!node) return;
+
+  uint8_t pubkeyhash[20] = {0};
+  if (!hdnode_get_ethereum_pubkeyhash(node, pubkeyhash)) {
```

Comment:
If no pubkey hash is returned, the device will just return to the home screen here. Is this a failure mode for the function? If so, should we use a call to `fsm_sendFailure()` to notify the upstream application?

Reply:
The failure message is initiated in fsm_getDeriveNode().